wlroots/util/region.py

```diff
@@ -0,0 +1,53 @@
+# Copyright (c) Matt Colligan 2021
+
+from typing import List
+
+from pywayland.protocol.wayland import WlOutput
+from wlroots import ffi, lib, Ptr
+from wlroots.wlr_types.box import Box
+
+
+class PixmanRegion32(Ptr):
+    def __init__(self) -> None:
+        """This is a convenience wrapper around pixman_region32_t
+
+        :param ptr:
+            The pixman_region32_t cdata pointer
+        """
+        self._ptr = ffi.new("struct pixman_region32 *")
+
+    def __enter__(self) -> "PixmanRegion32":
+        """Use the pixman_region32 in a context manager"""
+        lib.pixman_region32_init(self._ptr)
+        return self
+
+    def __exit__(self, exc_type, exc_value, exc_tb) -> None:
+        """Finish up when exiting the context"""
+        lib.pixman_region32_fini(self._ptr)
+
+    def rectangles_as_boxes(self) -> List[Box]:
+        nrects_ptr = ffi.new("int *")
+        rects = lib.pixman_region32_rectangles(self._ptr, nrects_ptr)
+        nrects = nrects_ptr[0]
+
+        boxes = []
+        for i in range(nrects):
+            boxes.append(
+                Box(rects.x1, rects.y1, rects.x2 - rects.x1, rects.y2 - rects.y1)
+            )
+            rects += 1
+        return boxes
+
+    def transform(
+        self, src: "PixmanRegion32", transform: WlOutput.transform, width: int, height: int
+    ) -> None:
+        """
+        Applies a transform to a region inside a box of size `width` x `height`.
+        """
+        lib.wlr_region_transform(self._ptr, src._ptr, transform, width, height)
+
+    def not_empty(self) -> int:
```

Comment:
Shouldn't this be a `bool` return type?

Reply:
Conceptually it is a boolean, but the source signature is an int. Does it need to match that? A `bool` makes more sense, and it'd just be a matter of wrapping the return value with `bool()`.

Reply:
Oh, it looks like the definition in the cffi definition is a boolean. Does it still come through as an int?

Reply:
Hmm, it seems I need to proof-read my PRs more closely. This is the original definition:
```c
typedef int pixman_bool_t;

PIXMAN_API
pixman_bool_t pixman_region32_not_empty (pixman_region32_t *region);
```
The return value is actually a bool because of the cffi signature, so I just need to change the type hint and problem solved.